docs/grammar.txt

```diff
@@ -7,7 +7,7 @@ ImportDirective = 'import' StringLiteral ('as' Identifier)? ';'
         | 'import' ('*' | Identifier) ('as' Identifier)? 'from' StringLiteral ';'
         | 'import' '{' Identifier ('as' Identifier)? ( ',' Identifier ('as' Identifier)? )* '}' 'from' StringLiteral ';'
 
-ContractDefinition = ( 'contract' | 'library' | 'interface' ) Identifier
+ContractDefinition = 'abstract'* ( 'contract' | 'library' | 'interface' ) Identifier
```

Comment:
Currently the parser is accepting `'abstract'*`. Should this be changed? What level of "abstraction" should be reflected with this `grammar.txt`? Just how the parser is basically implemented, or should this also reflect actual "semantics", e.g. that only contracts can be abstract. Then it would be probably something like

```
ContractDefinition = (('abstract'* 'contract') | 'library' | 'interface' ) Identifier
```

A question is also, whether `'abstract'*` is ok, or the parser should also be changed that only one or zero `abstract` tokens are accepted.

@chriseth

Reply:
@chriseth is `'abstract'*` ok?